reprospect/test/sass/instruction/address.py

```diff
@@ -1,149 +1,285 @@
 from __future__ import annotations
 
 import dataclasses
+import sys
 import typing
 
 import attrs
 import regex
 
+from reprospect.test.sass.instruction.memory import MemorySpace
 from reprospect.test.sass.instruction.pattern import PatternBuilder
 from reprospect.tools.architecture import NVIDIAArch
 
+if sys.version_info >= (3, 11):
+    from enum import StrEnum
+else:
+    from backports.strenum.strenum import StrEnum
+
+
+MODIFIER_STRIDE: typing.Final[str] = r'X(?:4|8)'
+"""Stride modifier."""
+
+class StrideModifier(StrEnum):
+    """
+    Stride modifier.
+    """
+    X4 = 'X4'
+    X8 = 'X8'
 
 @dataclasses.dataclass(frozen=True, slots=True)
 class AddressMatch:
     """
-    Result of matching an address operand.
+    Basic address operand, such as::
+
+        [R8+0xb5]
     """
     reg: str #: Register.
-    offset: str | None = None #: Offset.
-    ureg: str | None = None #: Uniform register holding the cache policy descriptor in a descriptor-based address.
+    offset: str | None = None #: Address offset.
 
     @classmethod
     def parse(cls, bits: regex.Match[str]) -> AddressMatch:
-        captured = bits.capturesdict()
-
-        if not (value := captured.get('reg')):
+        if not (reg := bits.group('reg')):
             raise ValueError(bits)
-        reg = value[0]
 
         offset: str | None = None
-        if value := captured.get('offset'):
+        if value := bits.capturesdict().get('offset'):
             if len(value) == 1:
                 offset = value[0]
 
-        ureg: str | None = None
-        if value := captured.get('ureg'):
+        return cls(reg=reg, offset=offset)
+
+@dataclasses.dataclass(frozen=True, slots=True)
+class GenericOrGlobalAddressMatch(AddressMatch):
+    """
+    Generic or global address operand, such as::
+
+        desc[UR42][R8+0xf1]
+    """
+    desc_ureg: str | None = None
+    """:py:attr:`RegisterType.UGPR` from cache policy descriptor."""
+
+    @classmethod
+    def parse(cls, bits: regex.Match[str]) -> GenericOrGlobalAddressMatch:
+        captured = bits.capturesdict()
+
+        base_match = AddressMatch.parse(bits)
+
+        desc_ureg: str | None = None
+        if value := captured.get('desc_ureg'):
             if len(value) == 1:
-                ureg = value[0]
+                desc_ureg = value[0]
 
         return cls(
-            reg=reg,
-            offset=offset,
-            ureg=ureg,
+            reg=base_match.reg,
+            offset=base_match.offset,
+            desc_ureg=desc_ureg,
+        )
+
+@dataclasses.dataclass(frozen=True, slots=True)
+class SharedAddressMatch(AddressMatch):
+    """
+    Shared address operand, such as::
+
+        [R25.X8+0x800]
+    """
+    stride: StrideModifier | None = None #: Stride modifier for shared memory addressing.
```

Comment:
```suggestion
    stride: StrideModifier | None = None
```

I guess if comments are too obvious, just get rid of them.